[INFRA-102] Adding archives.jenkins-ci.org

kohsuke opened the pull request:
... to serve historical archived releases. See [INFRA-102](https://issues.jenkins-ci.org/browse/INFRA-102) for the context.

kohsuke committed:
Provisioning LVM volume

kohsuke committed:
defining dependencies.

Looks like all the dependencies need to be specified manually. That's incredibly stupid.

kohsuke committed:
Deploy Apache

kohsuke committed:
present only touches /etc/fstab

kohsuke committed:
Enable rate limit module

kohsuke committed:
Moving the fragment into a separate file before adding more lines

kohsuke committed:
adding rate limit setting

kohsuke committed:
Can't figure out how to load a module file, switching to template

kohsuke committed:
Can't figure out how to make this dependency work

kohsuke committed:
mount task seems to require these arguments

Otherwise it ends up producing invalid /etc/fstab

kohsuke committed:
fixing the name based virtual host setting

kohsuke committed:
allow Jenkins to login as www-data to populate the releases

kohsuke committed:
Assign this directory to apache

kohsuke committed:
documenting the bandwidth throttling scheme

kohsuke committed:
disable directory indexing

kohsuke committed:
Light test as a touch up

kohsuke committed:
added serverspec test

kohsuke committed:
To test this better, added a fake block device during vagrant run.

LVM config is made customizable.

kohsuke committed:
Externalized LVM setting as hiera config

kohsuke committed:
These parameters are unused now

kohsuke committed:
Leaving a TODO comment

cloudbees-pull-request-builder commented:
[infra » jenkins-infra #168](https://jenkins.ci.cloudbees.com/job/infra/job/jenkins-infra/168/) FAILURE
Looks like there's a problem with this pull request

rtyler commented:
I'll take a look at this tomorrow, got caught up with work :(

kohsuke commented:
Retriggering the pull request build.

cloudbees-pull-request-builder commented:
[infra » jenkins-infra #170](https://jenkins.ci.cloudbees.com/job/infra/job/jenkins-infra/170/) FAILURE
Looks like there's a problem with this pull request

rtyler commented:
```
Failures:

  1) profile::archives should contain Package[apache2]
     Failure/Error: should contain_package('apache2')
       expected that the catalogue would contain Package[apache2]
     # ./spec/classes/profile/archives_spec.rb:6:in `block (2 levels) in <top (required)>
```

This looks like a legit test failure

kohsuke committed:
Updating the test

Looks like the apache module uses the name 'httpd' for everything but
set the name attribute differently.

Feels bit too much like testing the internals. Perhaps I should just
check contain_class("apache")

kohsuke commented:
My bad, that was a test problem.

cloudbees-pull-request-builder commented:
[infra » jenkins-infra #172](https://jenkins.ci.cloudbees.com/job/infra/job/jenkins-infra/172/) FAILURE
Looks like there's a problem with this pull request

kohsuke committed:
Here's hoping this would make lint happy

cloudbees-pull-request-builder commented:
[infra » jenkins-infra #174](https://jenkins.ci.cloudbees.com/job/infra/job/jenkins-infra/174/) FAILURE
Looks like there's a problem with this pull request

kohsuke commented:
@rtyler , while you review the change, please check the sanity of the expression `str2bool("${vagrant}")`

This is copied from a Google search result, and I don't understand why one just wouldn't simply do `str2bool($vagrant)`

kohsuke commented:
Not sure if failure in build #174 is transient. Trying it again.

cloudbees-pull-request-builder commented:
[infra » jenkins-infra #175](https://jenkins.ci.cloudbees.com/job/infra/job/jenkins-infra/175/) FAILURE
Looks like there's a problem with this pull request

rtyler commented on dist/profile/manifests/archives.pp:
```diff
@@ -0,0 +1,74 @@
+  }
+
+  # volume configuration is in hiera
+  include ::lvm
```
For consistency and style, includes should be at the top of the class. I'll tidy this up later

rtyler commented on dist/profile/manifests/archives.pp:
```diff
@@ -0,0 +1,74 @@
+    ensure => directory,
+  }
+
+  include apache
```
Ditto here

rtyler commented:
Mostly looks good to me, I'm going to merge this to `staging` and then do some minor non-functional cleanup before pushing to production later this afternoon.

I'm really impressed with this though!
